### DISCUSSION TITLE
Can a hash code be added to OTP?

### DISCUSSION THREAD

**Discussion by ursrudra** [appwrite/appwrite#5122] [Ideas]
Can a hash code be added to OTP?

**APP_HASH**
The Android app hash is very important, as the existence of this hash in the SMS Message is what the SMS Retriever API will look for. See Google's documentation on how to Compute your app's hash string. You can also find it under the Settings menu item in the sample application.

https://www.twilio.com/docs/sms/app-verification

**Comment by stnguyen90** [comment]
@ursrudra, thanks for raising this! Is this something that is specific to Twilio? Or do you know if other providers also support something like this?

**Reply by ursrudra** [reply]
@stnguyen90 Thank you for your response. yes, Android has this support through SMS Retriever API https://developers.google.com/identity/sms-retriever/overview

one possible solution is giving users a SMS body template  option and secret will be inserted into that. So that users by themselves can add the hashcode irrespective of SMS provider support.

Reference: Firebase authentication templates
